### PR TITLE
Require explicit nntile device for integer tensor inputs

### PR DESCRIPTION
**Pull request by Muxas**
## Summary

Enforces the policy that all NNTile kernel inputs must already be on `device="nntile"`. Host→nntile transfers are no longer implicit for integer/metadata tensors.

### Kernel input validation (C++)
- **Cross-entropy**: `target` must be on nntile (was CPU or nntile)
- **Embedding**: `indices` must be on nntile (forward and backward)
- **SDPA**: optional `mask` must be on nntile

### Python
- `torch_nntile.training.cross_entropy` rejects CPU targets instead of silently calling `.cpu()`
- `GPT2Model` creates causal masks on the same device as `input_ids`
- `train_full_batch_step` docstring documents the nntile input requirement

### Examples & tests
- GPT-2 examples and tests now use explicit `.to("nntile")` for `input_ids`, `labels`, `indices`, and masks
- `test_embedding_indices_on_cpu` replaced with `test_embedding_rejects_cpu_indices`
- Added `test_cross_entropy_rejects_cpu_target`

### Docs
- README updated: embedding indices, SDPA masks, and cross-entropy labels must be on nntile

## Migration

Before:
```python
labels = torch.randint(0, vocab_size, (batch, seq))  # CPU OK
loss = cross_entropy(logits_nnt, labels)
```

After:
```python
labels = torch.randint(0, vocab_size, (batch, seq)).to("nntile")
loss = cross_entropy(logits_nnt, labels)
```

Explicit `.to("nntile")` / `.cpu()` remain the supported transfer paths.

<div><a href="https://cursor.com/agents/bc-223282dc-9aa8-4799-a118-9a2b0e36758f"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-223282dc-9aa8-4799-a118-9a2b0e36758f"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

**Comment by Muxas** [comment]
@cursor review

**Comment by Muxas** [comment]
@cursor review
